draft-ietf-gnap-core-protocol.md

```diff
@@ -4703,6 +4712,59 @@ mechanism, but the AS also returns other proofing methods, then the AS
 will deny a request from that client instance using a different proofing
 mechanism.
 
+## RS-first Method of AS Discovery {#rs-request-without-token}
+
+If the client instance calls an RS without an access token, or with an
+invalid access token, the RS MAY respond to the client instance with an
+authentication header indicating that GNAP needs to be used
+to access the resource. The address of the GNAP
+endpoint MUST be sent in the "as_uri" parameter. The RS MAY
+additionally return a resource reference that the client instance MAY use in
+its access token request. This
+resource reference MUST be sufficient for at least the action
+the client instance was attempting to take at the RS and MAY be more
+powerful.
+The means for the RS to determine the resource reference are out of scope
+of this specification, but some dynamic methods are discussed in
+{{I-D.draft-ietf-gnap-resource-servers}}.
+The content of the resource handle is opaque to the client instance.
+
+~~~
+WWW-Authenticate: \
+  GNAP as_uri=https://server.example/tx,access=FWWIKYBQ6U56NL1
+~~~
+
+The client instance then makes a request to the "as_uri" as described in 
+{{request}}, with the value of "access" as one of the members
+of the `access` array in the `access_token` portion of the request. The
+client instance MAY request additional resources and other information.
+The client instance MAY request multiple access tokens.
+
+In this non-normative example, the client instance is requesting a single access 
+token using the resource reference `FWWIKYBQ6U56NL1` received from the RS
+in addition to the `dolphin-metadata` resource reference that the client instance
+has been configured with out of band. 
+
+~~~
+POST /tx HTTP/1.1
+Host: server.example.com
+Content-Type: application/json
+Detached-JWS: ejy0...
+
+{
+    "access_token": {
+        "access": [
+            "FWWIKYBQ6U56NL1",
+            "dolphin-metadata"
+        ]
+    },
+    "client": "KHRS6X63AJ7C7C4AZ9AO"
+}
+~~~
+
+If issued, the resulting access token would contain sufficient access to be used
```

Comment:
Seems ok, but we might be clearer of the "if issued" condition (again that's a decision made by the AS, which can use whichever rules make sense)

Reply:
This is echoing language that's used elsewhere in the spec, which was suggested during a text review. If it needs to be changed, we should change it consistently. I personally think it's fine as-is since it keeps things simple in the context -- we don't care how or why it was issued, just whether it was.

Reply:
Yes this can be checked globally at a later stage, just to make sure everything is consistent